### PR TITLE
PE-448: fee boost

### PR DESCRIPTION
**Pull request by arielmelendez**
Adds precise fee estimation to all existing writing commands and the ability to boost miner rewards by a multiple for all commands that send transactions. NOTE: the merge destination branch of `cli_v2_smartweave`. We'll want to merge to the `cli_v2` branch once the former has been merged there.

### CODE REVIEW

File: src/arfsdao.ts [hunk 1]
```diff
@@ -209,12 +217,13 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		super(arweave, appName, appVersion);
 	}
 
-	async createPublicFolder(
-		folderName: string,
-		driveId: DriveID,
-		parentFolderId?: FolderID,
+	async createPublicFolder({
+		folderData,
```
Comment (fedellen) [inline src/arfsdao.ts:221]: Interesting that only create folder re-uses the data object created for estimation.

**Review by fedellen — APPROVED** [review]
Looking good -- sent up a boosted drive this morning 🚀

File: src/arfs_trx_data_types.ts [hunk 2]
```diff
@@ -175,7 +186,7 @@ export class ArFSPrivateFileMetadataTransactionData extends ArFSFileMetadataTran
 				})
 			)
 		);
-		return new ArFSPrivateFileMetadataTransactionData(cipher, cipherIV, data);
+		return new ArFSPrivateFileMetadataTransactionData(cipher, cipherIV, fileKey, data);
```
Comment (arielmelendez) [inline src/arfs_trx_data_types.ts:189]: THESE PARAMS ARE OUT OF ORDER

Comment (arielmelendez) [reply src/arfs_trx_data_types.ts]: Handled in 913d2c8

**Review by vilenarios — APPROVED** [review]
The dry run looks straight forward enough to me.

FileKey also look to be used properly here.

Fee boosting is also seemingly correct and what I would expect.

**Review by agsuy — APPROVED** [review]

**Pull request merged**